File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/template/TemplateConcrete1.java
package jv.com.tencent.cain.design_pattern.template;

/**
 * @author : jiangyu
 * @date : 2019/11/3
 * @desc : 模版设计模式的子类
 */
public class TemplateConcrete1 extends TemplateAbstract {
    @Override
    void primitiveOperation1() {
        System.out.println("---TemplateConcrete1#primitiveOperation1---");
    }

    @Override
    void primitiveOperation2() {
        System.out.println("---TemplateConcrete1#primitiveOperation2---");
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/template/TemplateConcrete1.kt
package kt.com.tencent.cain.desigin_pattern.template

/**
 * @author : jiangyu
 * @date   : 2019/11/3
 * @desc   : 模版设计模式的子类
 */
class TemplateConcrete1 : TemplateAbstract() {
    override fun primitiveOperation1() {
        println("---TemplateConcrete1#primitiveOperation1---")
    }

    override fun primitiveOperation2() {
        println("---TemplateConcrete1#primitiveOperation2---")
    }

}